src/main/java/org/graylog/integrations/dataadapters/GreyNoiseCommunityIpLookupAdapter.java

```diff
@@ -0,0 +1,271 @@
+    private static Map<Object, Object> getResponseValueMap(Response response) {
+        Map<Object, Object> values;
+        try {
+            if (response.body() == null) {
+                values = Collections.emptyMap();
+            } else {
+                ObjectMapper mapper = new ObjectMapper();
+                TypeReference<Map<Object, Object>> ref = new TypeReference<Map<Object, Object>>() {};
+                values = mapper.readValue(response.body().byteStream(), ref);
+            }
+
+        } catch (IOException e) {
+            LOG.error("An error occurred while parsing parsing Lookup result. {}", e.getMessage(), e);
+            values = Collections.emptyMap();
+        }
+        return values;
+    }
+
+    public interface Factory extends LookupDataAdapter.Factory<GreyNoiseCommunityIpLookupAdapter> {
+
+        @Override
+        GreyNoiseCommunityIpLookupAdapter create(@Assisted("id") String id,
+                                                 @Assisted("name") String name,
+                                                 LookupDataAdapterConfiguration configuration);
+
+        @Override
+        Descriptor getDescriptor();
+
+    }
+
+    public static class Descriptor extends LookupDataAdapter.Descriptor<GreyNoiseCommunityIpLookupAdapter.Config> {
+
+        public Descriptor() {
+            super(ADAPTER_NAME, GreyNoiseCommunityIpLookupAdapter.Config.class);
+        }
+
+        @Override
+        public GreyNoiseCommunityIpLookupAdapter.Config defaultConfiguration() {
+            return GreyNoiseCommunityIpLookupAdapter.Config.builder()
+                    .type(ADAPTER_NAME)
+                    .apiToken(EncryptedValue.createUnset())
+                    .build();
+        }
+
+    }
+
+    @AutoValue
+    @JsonAutoDetect
+    @JsonDeserialize(builder = GreyNoiseCommunityIpLookupAdapter.Config.Builder.class)
+    @JsonTypeName(ADAPTER_NAME)
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public abstract static class Config implements LookupDataAdapterConfiguration {
+
+        @JsonProperty("api_token")
+        @NotEmpty
+        public abstract EncryptedValue apiToken();
+
+        public static GreyNoiseCommunityIpLookupAdapter.Config.Builder builder() {
+            return new AutoValue_GreyNoiseCommunityIpLookupAdapter_Config.Builder();
+        }
+
+        @AutoValue.Builder
+        public abstract static class Builder {
+            @JsonCreator
+            public static GreyNoiseCommunityIpLookupAdapter.Config.Builder create() {
+                return GreyNoiseCommunityIpLookupAdapter.Config.builder();
+            }
+
+            @JsonProperty(TYPE_FIELD)
+            public abstract GreyNoiseCommunityIpLookupAdapter.Config.Builder type(String type);
+
+            @JsonProperty("api_token")
+            public abstract GreyNoiseCommunityIpLookupAdapter.Config.Builder apiToken(EncryptedValue apiToken);
+
+            public abstract GreyNoiseCommunityIpLookupAdapter.Config build();
```

Comment:
```suggestion
        public abstract static class Builder {
            @JsonCreator
            public static Builder create() {
                return builder();
            }

            @JsonProperty(TYPE_FIELD)
            public abstract Builder type(String type);

            @JsonProperty("api_token")
            public abstract Builder apiToken(EncryptedValue apiToken);

            public abstract GreyNoiseCommunityIpLookupAdapter.Config build();
```
Very minor, but I think this is the extra `GreyNoiseCommunityIpLookupAdapter.Config` qualifier is not needed. Might be a little more readable without it.